apps/backend/tests/test_auth/test_skip_auth_guard.py

```
@@ -0,0 +1,274 @@
+    def test_dummy_credentials_rejected_when_node_env_is_production(self, monkeypatch):
+        monkeypatch.setenv("ENVIRONMENT", "development")
+        monkeypatch.setenv("NODE_ENV", "production")
+        monkeypatch.setenv("AWS_ACCESS_KEY_ID", "placeholder")
+        monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", "placeholder")
+
+        from app.config import Settings
+
+        with pytest.raises(ValueError, match="[Dd]ummy credential"):
+            Settings(
+                AWS_ACCESS_KEY_ID="placeholder",
+                AWS_SECRET_ACCESS_KEY="placeholder",
```

Ruff (0.15.15)

[error] Possible hardcoded password assigned to argument: "AWS_SECRET_ACCESS_KEY"

(S106)